Version 2.0

Davidmattei opened the pull request:
|Q              |A  |
|---------------|---|
|Bug fix?       |yes|
|New feature?   |yes|	
|BC breaks?     |no|	
|Deprecations?  |no|	
|Fixed tickets  |no|	

Requires new releases:
- CommonBundle: https://github.com/ems-project/EMSCommonBundle/pull/97
- FormBundle: https://github.com/ems-project/EMSFormBundle/pull/84

Davidmattei committed:
maintain(folder structer): create src folder (#30)

dameert commented:
Update this line in phpstan: 

        - %currentWorkingDirectory%/DependencyInjection/Configuration.php

To add the /src subfolder

Davidmattei committed:
Setup first test suite (#32)

Davidmattei committed:
maintain(composer): update deps (#33)

Remove client-helper-bundle
Remove symfony form, formBundle has this dep

dameert commented on .gitignore:
```diff
@@ -1,6 +1,3 @@
+.phpunit.result.cache
 composer.lock
-phpunit.xml
```
See comment on #32 /phpunit.xml is the local version of /phpunit.xml.dist

dameert commented on .gitignore:
```diff
@@ -1,6 +1,3 @@
+.phpunit.result.cache
 composer.lock
-phpunit.xml
```
See comment on #32 /phpunit.xml is the local version of /phpunit.xml.dist

Davidmattei replied:
Fixed in #34

Davidmattei committed:
maintain(test): Email handler (#34)

Davidmattei committed:
Phpcs and gitignore (#35)

Commit:
run test at the end

Commit:
maintain(test): serviceNow

Davidmattei commented:
96% coverage 👍

dameert commented on src/FormConfig/EmailConfig.php:
```diff
@@ -22,8 +22,8 @@ public function __construct(RenderedSubmission $submission)
 
-        $this->from = $message['from'];
-        $this->subject = $message['subject'];
+        $this->from = $message['from'] ?? 'noreply@elasticms.eu';
```
do we want our hosting to be the default email address for everybody using the project?

Davidmattei replied:
Fix https://github.com/ems-project/EMSSubmissionBundle/pull/31/commits/691e4279d3dd93797c55d50ef6d6b6ab49d97ded

dameert commented on tests/Functional/Handler/EmailHandlerTest.php:
```diff
@@ -0,0 +1,188 @@
+        ]];
+        $form = $this->formFactory->createBuilder(FormType::class, $data, [])
+            ->add('name', TextType::class)
+            ->add('files', FileType::class, ['multiple' => true])
```
Should we not use our components in testing in stead of the Symfony components directly?

Davidmattei replied:
No that's the responsibility of the form bundle

Commit:
feat(email): required from in endpoint

If the from email is not defined we throw an exception

Davidmattei committed:
maintain(php-cs-fixer): add php cs fixer and travis (#36)

Davidmattei committed:
Refactor and cleanup (#37)

Davidmattei committed:
feat(pdf hander): Add pdf handler (#38)

Refactor tests
Make handle responses better
Implement refactor on formBundle
Test coverage 98%

Commit:
ref(coverage): Fix tests and last remarks

100% code coverage

Commit:
Add missing pdf readme

Commit:
fix readme following -> chained
